src/dotnet/Fable.Compiler/CLI/Agent.fs

```diff
@@ -224,14 +217,24 @@ let startCompilation (msgHandler: Server.MessageHandler) (com: Compiler) (projec
                 // Don't send dependencies to JS client (see #1241)
                 project.AddDependencies(com.CurrentFile, babel.Dependencies)
                 Babel.Program(babel.FileName, babel.Body, babel.Directives, com.GetFormattedLogs())
-                |> toJson msgHandler
+                |> respond
         with ex ->
-            sendError msgHandler ex
+            sendError respond ex
     } |> Async.Start
 
-let startAgent () = MailboxProcessor<Server.MessageHandler>.Start(fun agent ->
+let startAgent () = MailboxProcessor<AgentMsg>.Start(fun agent ->
     let rec loop (checker: FSharpChecker) (state: Map<string,Project>) = async {
-        use! msgHandler = agent.Receive()
+      match! agent.Receive() with
+      | Respond(value, msgHandler) ->
+        msgHandler.Respond(fun writer ->
+            // Don't use `use` (pun unintended), it will close the underlying writer
+            let jsonWriter = new JsonTextWriter(writer)
```

Comment:
What about this?

``` fsharp
use writer = new JsonTextWriter (writer, CloseOutput=false)
```

Looks like the `JsonTextWriter` [returns a buffer](https://github.com/JamesNK/Newtonsoft.Json/blob/44539a8f002bc4d6dae919a0340531661c9bdd5d/Src/Newtonsoft.Json/JsonTextWriter.cs#L187) back to a pool on `Close`, which is called by `Dispose`.

Reply:
Good point! Let's do that, thank you!